### ISSUE TITLE
BeginsWith filter produces incorrect query

### ISSUE THREAD

**Issue by thebogusman** [AllocZero/EfficientDynamoDb#123]
Having a condition built like that: 
```
var filter = Condition.ForEntity<Note>();
var condition = filter.On(e => e.Pk).EqualTo(pk)
    & filter.On(e => e.Sk).BeginsWith(prefix);
```

produces the following query:
```
{
   "KeyConditionExpression":"(#f0 = :v0)AND(begins_with(#f1,:v1))",
   "ExpressionAttributeNames":{
      "#f0":"PK",
      "#f1":"SK"
   },
   "ExpressionAttributeValues":{
      ":v0":{
         "S":"CUSTOMER#00000001"
      },
      ":v1":"CONTEXT#00000002"
   },
   "TableName":"notes-table"
}
```

which gets rejected by the API as a 400 bad request, with the following error message: "Unexpected value type in payload".

It looks like the value for the parameter v1 is passed a string instead of a json object with a type.

As a workaround I'm using Between filter instead,

**Comment by lezzi** [comment]
Thank you for reporting the bug, @thebogusman . The fix is available in the `0.9.2-alpha.0.6` nuget version.

**Comment by thebogusman** [comment]
Amazing, it works correctly now. Thank you!
